multinet/db.py

```diff
@@ -114,18 +113,17 @@ def workspace_exists_internal(name: str) -> bool:
     return sysdb.has_database(name)
 
 
-def create_workspace(name: str) -> None:
+def create_workspace(name: str, ws_doc: Workspace) -> None:
     """Create a new workspace named `name`."""
 
     if workspace_exists(name):
         raise AlreadyExists("Workspace", name)
 
     coll = workspace_mapping_collection()
-    new_doc = {"name": name, "internal": generate_arango_workspace_name()}
-    coll.insert(new_doc)
+    coll.insert(ws_doc)
 
     try:
-        db("_system").create_database(new_doc["internal"])
+        db("_system").create_database(ws_doc["internal"])
     except DatabaseCreateError:
         # Could only happen if there's a name collisison
         raise InternalServerError()
```

Comment:
This may not be strictly related to this PR, but I think here we should wait to insert the workspace document into the mapping collection until after the database creation has succeeded. The way that things are currently implemented, we could have dangling entries in the workspace mapping collection. I'm okay with addressing this in a separate issue if that's what we want to do.

Reply:
See 255bfb6d356cc60a89f78d13e8d808ab82d49189.

In general, we may want to write a sort of fsck script to validate our server data state. See #398.